### PR TITLE
Add has_hispanic_latino_ethnicity to dim_student

### PR DESCRIPTION
**Pull request by ryanaguilar**
## Description & motivation:
This PR would add the has_hispanic_latino_ethnicity field to dim_student.  Addresses a request from the [Boston Stadium implementation](https://educationanalytics.monday.com/boards/2610522828/pulses/5838180248) and is an item on the [EDU Project Development](https://educationanalytics.monday.com/boards/3556827529/pulses/6109566542) board.

## PR Merge Priority:
- [ ] Low </br>
- [x] Medium </br>
- [ ] High </br>

## Changes to existing models:
In edu_wh
- `bld_ef3__stu_race_ethnicity` : Bring has_hispanic_latino_ethnicity column in from stg_ef3__student_education_organization_associations
- `bld_ef3__immutable_stu_demos` : Bring has_hispanic_latino_ethnicity column in from bld_ef3__stu_race_ethnicity
- `dim_student`: Bring has_hispanic_latino_ethnicity column in from bld_ef3__immutable_stu_demos

## Tests and QC done:
- Executed a dbt project run and ensured it was successful.

### CODE REVIEW

File: models/core_warehouse/dim_student.yml [hunk 1]
```diff
@@ -71,4 +71,6 @@ models:
       - name: race_array
         description: A list of the student's races.
       - name: safe_display_name
-        description: Display name with student ID, safe for use in grouping operations in a BI tool, to avoid combining students with the same name.
\ No newline at end of file
+        description: Display name with student ID, safe for use in grouping operations in a BI tool, to avoid combining students with the same name.
+      - name: has_hispanic_latino_ethnicity
+        description: The student's status for having Hispanic/Latino ethnicity
```
Comment (rlittle08) [inline models/core_warehouse/dim_student.yml:76]: we could use Ed-Fi's definition here. I usually do that just so I don't have to come up with one on my own, but what you have is fine too, this is a self-explanatory column

File: models/core_warehouse/dim_student.sql [hunk 2]
```diff
@@ -188,7 +188,8 @@ formatted as (
         stg_student.api_year = max(stg_student.api_year) over(partition by stg_student.k_student_xyear) as is_latest_record,
        
         stu_immutable_demos.race_array,
-        stu_immutable_demos.safe_display_name
+        stu_immutable_demos.safe_display_name,
+        stu_immutable_demos.has_hispanic_latino_ethnicity
```
Comment (rlittle08) [inline models/core_warehouse/dim_student.sql:192]: Thoughts on moving the col to be right after "race_ethnicity"? users might not think it exists if it's the last column, I think we generally try to put the least human-used columns at the end

**Review by rlittle08 — COMMENTED** [review]
looks good, I checked some edge cases in dev_ra_wh 👍 

just left a couple tiny suggestions

File: models/core_warehouse/dim_student.yml (hunk 1 above)
Comment (ryanaguilar) [reply models/core_warehouse/dim_student.yml:76]: That makes sense. Ed-Fi's definintion from the data handbook?

File: models/core_warehouse/dim_student.sql (hunk 2 above)
Comment (ryanaguilar) [reply models/core_warehouse/dim_student.sql:192]: Agreed, right after "race_ethnicity" makes the most sense.  Based on the other columns I thought maybe the columns were in the order they were added to the table.

Comment (ryanaguilar) [reply models/core_warehouse/dim_student.sql]: e16118c32d2b8f2bc430957031e0d693d5bc0b0c

File: models/core_warehouse/dim_student.yml (hunk 1 above)
Comment (rlittle08) [reply models/core_warehouse/dim_student.yml:76]: yeah, meant to link this but forgot https://schema.ed-fi.org/datahandbook-v400/#/HispanicLatinoEthnicity1653

Comment (ryanaguilar) [reply models/core_warehouse/dim_student.yml]: b3d7c0acec73c874d8d34f1231b5e47edac61567

**Review by rlittle08 — APPROVED** [review]

**Pull request merged**